src/core/listener.c

```diff
@@ -122,27 +155,28 @@ MsQuicListenerClose(
 
 #pragma prefast(suppress: __WARNING_25024, "Pointer cast already validated.")
     QUIC_LISTENER* Listener = (QUIC_LISTENER*)Handle;
-    QUIC_REGISTRATION* Registration = Listener->Registration;
 
-    //
-    // Make sure the listener has unregistered from the binding.
-    //
-    MsQuicListenerStop(Handle);
+    QUIC_LIB_VERIFY(!Listener->AppClosed);
+    Listener->AppClosed = TRUE;
 
-    CxPlatRundownUninitialize(&Listener->Rundown);
+    if (Listener->StopCompleteThreadID == CxPlatCurThreadID()) {
```

Comment:
I'm not quite sure on this. What happens if the app calls ListenerClose() from a different callback?

Reply:
There's nothing we can do unless really.